ethcore/src/blockchain/block_info.rs

```diff
@@ -40,12 +42,55 @@ pub enum BlockLocation {
 	/// It's part of the fork which should become canon chain,
 	/// because it's total difficulty is higher than current
 	/// canon chain difficulty.
-	BranchBecomingCanonChain {
-		/// Hash of the newest common ancestor with old canon chain.
-		ancestor: H256,
-		/// Hashes of the blocks between ancestor and this block.
-		enacted: Vec<H256>,
-		/// Hashes of the blocks which were invalidated.
-		retracted: Vec<H256>,
+	BranchBecomingCanonChain(BranchBecomingCanonChainData),
```

Comment:
need to move it to seraparate struct `BranchBecomingCanonChainData` to represent it later as a tuple to use in generic serializer 
`impl<'a, V1, V2, T3> ToBytesWithMap for (&'a Vec<V1>, &'a Vec<V2>, &'a T3)`

otherwise you need to write full serialization for the whole struct `BlockLocation` by hand